File: manual/data/cpp/tests/test_pyphonetics_phonetics.cpp
#include <gtest/gtest.h>
#include "../src/function_pyphonetics_phonetics.cpp"

class TestPhoneticProcessor : public ::testing::Test {
protected:
    void SetUp() override {
        processor = new PhoneticProcessor();
    }
    void TearDown() override {
        delete processor;
    }
    PhoneticProcessor* processor;
};

TEST_F(TestPhoneticProcessor, BasicString) {
    EXPECT_EQ(processor->generate_soundex("hello"), "H400");
}

TEST_F(TestPhoneticProcessor, SameSound) {
    EXPECT_EQ(processor->generate_soundex("halo"), "H400");
}

TEST_F(TestPhoneticProcessor, DifferentSound) {
    EXPECT_EQ(processor->generate_soundex("world"), "W643");
}

TEST_F(TestPhoneticProcessor, EmptyString) {
    EXPECT_EQ(processor->generate_soundex(""), "The given string is empty.");
}

TEST_F(TestPhoneticProcessor, SingleCharacter) {
    EXPECT_EQ(processor->generate_soundex("a"), "A000");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/java/tests/test_pyphonetics_phonetics.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_pyphonetics_phonetics {
    private function_pyphonetics_phonetics processor;

    @BeforeEach
    void setUp() {
        processor = new function_pyphonetics_phonetics();
    }

    @Test
    void test_basic_string() {
        String result = processor.generate_soundex("hello");
        assertEquals("H400", result);
    }

    @Test
    void test_same_sound() {
        String result = processor.generate_soundex("halo");
        assertEquals("H400", result);
    }

    @Test
    void test_different_sound() {
        String result = processor.generate_soundex("world");
        assertEquals("W643", result);
    }

    @Test
    void test_empty_string() {
        String result = processor.generate_soundex("");
        assertEquals("The given string is empty.", result);
    }

    @Test
    void test_single_character() {
        String result = processor.generate_soundex("a");
        assertEquals("A000", result);
    }
}